autosklearn/util/backend.py

```diff
@@ -61,20 +77,39 @@ def _prepare_directories(self, temporary_directory, output_directory):
             else '/tmp/autosklearn_output_%d_%d' % (pid, random_number)
 
     def create_directories(self):
-        try:
+        if self.shared_mode:
+            # If shared_mode == True, the tmp and output dir will be shared
+            # by different instances of auto-sklearn.
+            try:
+                os.makedirs(self.temporary_directory)
+            except OSError:
+                pass
+            try:
+                os.makedirs(self.output_directory)
+            except OSError:
+                pass
+
+        else:
+            # Exception is raised if self.temporary_directory already exists.
             os.makedirs(self.temporary_directory)
-        except OSError:
-            pass
-        try:
+            self._tmp_dir_created = True
+
+            # Exception is raised if self.output_directory already exists.
             os.makedirs(self.output_directory)
-        except OSError:
-            pass
+            self._output_dir_created = True
+
 
     def __del__(self):
         self.delete_directories(force=False)
 
     def delete_directories(self, force=True):
         if self.delete_output_folder_after_terminate or force:
+            if self._output_dir_created is False and self.shared_mode is False:
+                raise OSError("Failed to delete output dir: %s "
```

Comment:
Please do not raise `OSError` here as this is not raised by the operating system. `ValueError` should be fine.